### PR TITLE
Resolving symlinks, fixed breadcrumb for parent folder, added NodeFinder

### PR DESCRIPTION
**Pull request by VladMartyshevskyi**
Resolving symlinks
Fixed breadcrumb for the parent folder
Fixed searching nodes (used NodeFinder)

### CODE REVIEW

File: webapp/src/main/webapp/WEB-INF/conf/officeonline/configuration.xml [hunk 1]
```diff
@@ -929,4 +933,31 @@
...
+  <external-component-plugins>
+    <target-component>org.exoplatform.services.jcr.RepositoryService</target-component>
+    <component-plugin>
+      <name>add.namespaces</name>
+      <set-method>addPlugin</set-method>
+      <type>org.exoplatform.services.jcr.impl.AddNamespacesPlugin</type>
+      <init-params>
+        <properties-param>
+          <name>namespaces</name>
+          <property name="oo" value="http://www.exoplatform.com/jcr/officeonline/1.0/" />
```
Comment (pnedonosko) [inline webapp/src/main/webapp/WEB-INF/conf/officeonline/configuration.xml]: `oo` doesn't tell clearly what is it, let's use more obvious name `msoffice`

File: services/src/main/resources/conf/portal/jcr/officeonline-nodetypes.xml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  You should have received a copy of the GNU Lesser General Public
+  License along with this software; if not, write to the Free
+  Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+  02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+<nodeTypes xmlns:nt="http://www.jcp.org/jcr/nt/1.0" xmlns:mix="http://www.jcp.org/jcr/mix/1.0"
+  xmlns:jcr="http://www.jcp.org/jcr/1.0">
+
+
+  <nodeType name="oo:filePreferences" isMixin="false" hasOrderableChildNodes="true" primaryItemName="">
```
Comment (pnedonosko) [inline services/src/main/resources/conf/portal/jcr/officeonline-nodetypes.xml]: it's general nodetype, why not move it to ECMS?

File: services/src/main/resources/conf/portal/jcr/officeonline-nodetypes.xml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      <childNodeDefinition name="*" defaultPrimaryType="" autoCreated="false" mandatory="false"
+        onParentVersion="VERSION" protected="false" sameNameSiblings="false">
+        <requiredPrimaryTypes>
+          <requiredPrimaryType>nt:base</requiredPrimaryType>
+        </requiredPrimaryTypes>
+      </childNodeDefinition>
+    </childNodeDefinitions>
+  </nodeType>
+
+  <nodeType name="oo:userPreferences" isMixin="false" hasOrderableChildNodes="true" primaryItemName="">
```
Comment (pnedonosko) [inline services/src/main/resources/conf/portal/jcr/officeonline-nodetypes.xml]: it's general nodetype, let's not move it to ECMS

File: services/src/main/resources/conf/portal/jcr/officeonline-nodetypes.xml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      <childNodeDefinition name="*" defaultPrimaryType="" autoCreated="false" mandatory="false"
+        onParentVersion="VERSION" protected="false" sameNameSiblings="false">
+        <requiredPrimaryTypes>
+          <requiredPrimaryType>nt:base</requiredPrimaryType>
+        </requiredPrimaryTypes>
+      </childNodeDefinition>
+    </childNodeDefinitions>
+  </nodeType>
+  
+  <nodeType name="oo:officeonlineFile" isMixin="true" hasOrderableChildNodes="false">
```
Comment (pnedonosko) [inline services/src/main/resources/conf/portal/jcr/officeonline-nodetypes.xml]: don't repeat yourself, if namespace prefix already about Office Online `oo`, then don't tell this `officeonline` in the type name.

**Review by pnedonosko — CHANGES_REQUESTED** [review]

File: services/src/main/resources/conf/portal/jcr/officeonline-nodetypes.xml (hunk 4 above)
Comment (VladMartyshevskyi) [reply services/src/main/resources/conf/portal/jcr/officeonline-nodetypes.xml]: Fixed: https://github.com/exo-addons/office-online/pull/18/commits/34bebb38165ffcb6c2e89d0bf3a98b8d041d7593

File: services/src/main/resources/conf/portal/jcr/officeonline-nodetypes.xml (hunk 3 above)
Comment (VladMartyshevskyi) [reply services/src/main/resources/conf/portal/jcr/officeonline-nodetypes.xml]: Fixed: https://github.com/exo-addons/office-online/pull/18/commits/34bebb38165ffcb6c2e89d0bf3a98b8d041d7593

File: services/src/main/resources/conf/portal/jcr/officeonline-nodetypes.xml (hunk 2 above)
Comment (VladMartyshevskyi) [reply services/src/main/resources/conf/portal/jcr/officeonline-nodetypes.xml]: Fixed: https://github.com/exo-addons/office-online/pull/18/commits/34bebb38165ffcb6c2e89d0bf3a98b8d041d7593

File: webapp/src/main/webapp/WEB-INF/conf/officeonline/configuration.xml (hunk 1 above)
Comment (VladMartyshevskyi) [reply webapp/src/main/webapp/WEB-INF/conf/officeonline/configuration.xml]: Fixed: https://github.com/exo-addons/office-online/pull/18/commits/34bebb38165ffcb6c2e89d0bf3a98b8d041d7593

**Comment by VladMartyshevskyi** [comment]
This PR should be merged after https://github.com/exodev/ecms/pull/512 approved

**Pull request merged**